index_copy.js

```diff
@@ -0,0 +1,104 @@
+const battleship = () => {
+
+    // create objects with the given parametrs
+    let playerOne = {
+      name: "",
+      shipCount: 0,
+      gameBoard: []
+    }
+    let playerTwo = {
+      name: "",
+      shipCount: 0,
+      gameBoard: []
+    };
+  
+    // asking for the players' name
+    playerOne.name = prompt("Player 1 what is your name?");
+    playerTwo.name = prompt("Player 2 what is your name?");
+  
+    // creating game boards
+    let boardColumn = 4;
+    let shipCountFinal = 4;
+  
+    // add game board with 0
+    for (let i = 0; i < boardColumn; i++) {
+      let boardRow = []
+      for (let j = 0; j < boardColumn; j++) {
+        boardRow.push(0)
+      }
+  
+  
+  // add rows to the end of an empty list
+      playerOne.gameBoard.push(boardRow);
+      playerTwo.gameBoard.push(boardRow);
+    }
+    console.log("Test", playerTwo.gameBoard);
```

Comment:
you can fix the problem, by making the building part of your code by a fucntion that you can run twice, once for each player. By running the function twice you are creating new arrays in memory, so that the players aren't getting the reference to the same array.
```
    const buildBoard = player => {
      for (let i = 0; i < boardColumn; i++) {
        let boardRow = []
        for (let j = 0; j < boardColumn; j++) {
          boardRow.push(0)
        }
        // add rows to the end of an empty list
        player.gameBoard.push(boardRow);
      }
    }

    buildBoard (playerOne);
    buildBoard (playerTwo);
```

Reply:
@marymatt504, thanks for your help. I redid this part of code using for loop and it works as expected now